docs/source/ucx-debug.rst

```diff
@@ -0,0 +1,71 @@
+UCX Debugging
+=============
+
+
+
+System Configuration
+--------------------
+
+``ibdev2netdev`` -- check to ensure at least one IB controller is configured for IPoIB
+
+::
+    user@mlnx:~$ ibdev2netdev
+    mlx5_0 port 1 ==> ib0 (Up)
+    mlx5_1 port 1 ==> ib1 (Up)
+    mlx5_2 port 1 ==> ib2 (Up)
+    mlx5_3 port 1 ==> ib3 (Up)
+
+
+ucx_perftest
+------------
+
+Perf test for IB should be in the 10+ GB/s range
+::
+
+    CUDA_VISIBLE_DEVICES=0 UCX_NET_DEVICES=mlx5_0:1 UCX_TLS=rc,cuda_copy ucx_perftest -t tag_bw -m cuda -s 10000000 -n 10 -p 9999 & \
+    CUDA_VISIBLE_DEVICES= 1UCX_NET_DEVICES=mlx5_1:1 UCX_TLS=rc,cuda_copy ucx_perftest `hostname` -t tag_bw -m cuda -s 100000000 -n 10 -p 9999
+
+    +--------------+-----------------------------+---------------------+-----------------------+
+    |              |       latency (usec)        |   bandwidth (MB/s)  |  message rate (msg/s) |
+    +--------------+---------+---------+---------+----------+----------+-----------+-----------+
+    | # iterations | typical | average | overall |  average |  overall |   average |   overall |
+    +--------------+---------+---------+---------+----------+----------+-----------+-----------+
+    +------------------------------------------------------------------------------------------+
+    | API:          protocol layer                                                             |
+    | Test:         tag match bandwidth                                                        |
+    | Data layout:  (automatic)                                                                |
+    | Send memory:  cuda                                                                       |
+    | Recv memory:  cuda                                                                       |
+    | Message size: 100000000                                                                  |
+    +------------------------------------------------------------------------------------------+
+                10     0.000  9104.800  9104.800   10474.41   10474.41         110         110
+
+
+Perf test for NVLink should be in the 20+ GB/s range
+::
+
+    CUDA_VISIBLE_DEVICES=0 UCX_TLS=cuda_ipc,cuda_copy,tcp,sockcm UCX_SOCKADDR_TLS_PRIORITY=sockcm  ucx_perftest -t tag_bw -m cuda -s 10000000 -n 10 -p 9999 -c 0 & \
+    CUDA_VISIBLE_DEVICES=1 UCX_TLS=cuda_ipc,cuda_copy,tcp,sockcm UCX_SOCKADDR_TLS_PRIORITY=sockcm ucx_perftest `hostname` -t tag_bw -m cuda -s 100000000 -n 10 -p 9999 -c 1
```

Comment:
`-c` is NUMA dependent, so you should be explicit in specifying that the user must choose a CPU that matches the CPU Affinity for that GPU, as reported by `nvidia-smi topo -m`.

Reply:
This is resolved in https://github.com/rapidsai/ucx-py/pull/624/commits/8eb2d479ed94627ce7cb13f36a1df55871ff8a7c .